src/object/obj_class.c

```diff
@@ -328,3 +356,127 @@ daos_oclass_grp_nr(struct daos_oclass_attr *oc_attr, struct daos_obj_md *md)
 	/* NB: @md is unsupported for now */
 	return oc_attr->ca_grp_nr;
 }
+
+/** a structure to map EC object class to gf tables */
+struct daos_oc_gftbls {
+	/** object class id */
+	daos_oclass_id_t	 e_oc_id;
+	/**
+	 * pointer to array of input tables generated from coding coefficients.
+	 */
+	unsigned char		*e_gftbls;
+};
+
+static struct daos_oc_gftbls	*daos_oc_gftbls_map;
+static int			 daos_oc_gftbls_nr;
+
+void
+daos_oc_gftbls_map_fini()
+{
+	struct daos_oc_gftbls	*oc_gftbls;
+	struct daos_obj_class	*oc;
+	int			 ocnr = 0;
+	int			 i;
+
+	if (daos_oc_gftbls_map == NULL)
+		return;
+
+	for (oc = &daos_obj_classes[0]; oc->oc_id != DAOS_OC_UNKNOWN; oc++) {
+		if (oc->oc_attr.ca_resil == DAOS_RES_EC)
+			ocnr++;
+	}
+	D_ASSERT(daos_oc_gftbls_nr == ocnr);
+
+	for (i = 0; i < ocnr; i++) {
+		oc_gftbls = &daos_oc_gftbls_map[i++];
+		if (oc_gftbls->e_gftbls != NULL)
+			D_FREE(oc_gftbls->e_gftbls);
+	}
+
+	D_FREE(daos_oc_gftbls_map);
+	daos_oc_gftbls_map = NULL;
+	daos_oc_gftbls_nr = 0;
+}
+
+int
+daos_oc_gftbls_map_init()
+{
+	struct daos_oc_gftbls	*oc_gftbls;
+	struct daos_obj_class	*oc;
+	unsigned char		*encode_matrix = NULL;
+	int			 ocnr = 0;
+	int			 i;
+	int			 k, p, m;
+	int			 rc;
+
+	if (daos_oc_gftbls_map != NULL)
+		return 0;
+
+	for (oc = &daos_obj_classes[0]; oc->oc_id != DAOS_OC_UNKNOWN; oc++) {
+		if (oc->oc_attr.ca_resil == DAOS_RES_EC)
+			ocnr++;
+	}
+
+	if (ocnr == 0)
+		return 0;
+
+	D_ALLOC_ARRAY(daos_oc_gftbls_map, ocnr);
+	if (daos_oc_gftbls_map == NULL)
+		D_GOTO(failed, rc = -DER_NOMEM);
+
+	i = 0;
+	for (oc = &daos_obj_classes[0]; oc->oc_id != DAOS_OC_UNKNOWN; oc++) {
+		if (oc->oc_attr.ca_resil != DAOS_RES_EC)
+			continue;
+
+		oc_gftbls = &daos_oc_gftbls_map[i++];
+		oc_gftbls->e_oc_id = oc->oc_id;
+		k = oc->oc_attr.u.ec.e_k;
+		p = oc->oc_attr.u.ec.e_p;
+		m = k + p;
+		D_ALLOC(oc_gftbls->e_gftbls, k * p * 32);
```

Comment:
what is the 32 at here?

Reply:
It's a constant value require when allocating the GF table. GF table size is <number of data cells> * <number of code cells> * <space needed for data generated for each input coefficient>

Reply:
[number of data cells] * [number of code cells] * [space needed for data generated for each input coefficient]

Reply:
yes, thanks Joe. 
BTW, by reading some isa-l code and some web pages, I think the variable name of k/p/m/len is pretty well known accustomed usage in EC just means the number of data/parity/total stripe/cell and length of each. it looks good to me to keep that usage as far as possible in the code.

Reply:
I like k, p, m, and len (this is ISA-L,, but not entirely standard).  I used longer names to be more descriptive (len could be confused), but I agree, so let's go with these.